tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunCtorRule.scala

```diff
@@ -68,10 +71,15 @@ class FunCtorRule(rewriter: SymbStateRewriter) extends RewritingRule {
 
     // associate a value of the uninterpreted function with a cell
     def addCellCons(domElem: ArenaCell, relElem: ArenaCell): Unit = {
-      val inDomain = tla.in(domElem.toNameEx, domainCell.toNameEx).typed(BoolT1())
-      val inRelation = tla.in(relElem.toNameEx, relation.toNameEx).typed(BoolT1())
-      val iff = tla.equiv(inDomain, inRelation).typed(BoolT1())
-      rewriter.solverContext.assertGroundExpr(iff)
+      val inDomain = inOperFactory.mkAccessOp(domElem, domainCell).typed(BoolT1())
+      val inRelation = inOperFactory.mkUpdateOp(relElem, relation).typed(BoolT1())
+      val expr = if (rewriter.solverContext.config.smtEncoding == "arrays") {
+        val notInRelation = inOperFactory.mkUnchangedOp(relElem, relation).typed(BoolT1())
```

Comment:
Exactly because they are empty by default. We need `notInRelation` to keep track of the SSA representation of the array. Will add a comment as requested.